apps/google-analytics-4/frontend/src/components/main-app/constants/noteMessages.ts

```diff
@@ -14,3 +16,20 @@ export const getContentTypeSpecificMsg = (contentTypeName: string) => ({
   noSlugContentMsg: `This ${contentTypeName} entry does not have a valid slug field. Please add a field of type short text to this entry and configure it on the app configuration page.`,
   notPublishedMsg: `This ${contentTypeName} entry has not yet been published. Please publish the entry to view this Google Analytics 4 sidebar app.`,
 });
+
+export const getAppInstallationParamsMsg = (
+  serviceAccountKeyId: ServiceAccountKeyId | undefined,
+  propertyId: string | undefined
+) => {
+  const serviceAccountMsg = 'Service Account Key';
+  const propertyMsg = 'Google Analytics 4 property';
+  let missingParams = '';
+
+  if (!serviceAccountKeyId && !propertyId)
+    missingParams = `${serviceAccountMsg} and ${propertyMsg}`;
+  else {
```

Comment:
I'm assuming what you're trying to do here is rendering the appropriate message when both the `serviceAccountId` and `propertyId` is missing or if one of them is missing.

But your else logic here assumes that at least one of the two are always missing. For example, what if there is a `propertyId`? Then it would render `(missingParams = propertyMsg);` still.

So it's one of two things
1. We always assume that at least one of the two properties are missing, but it's not exactly clear from the code that this is the assumption.
2. This is a bug in the code in the case where there is a propertyId

Reply:
Actually I think this is best without investing too much time. Just do if checks forget using elses and then just add a comment that one of the two params are guaranteed to be null. Also I would adjust the function name to be more in tune with the code logic.

```
  if (!serviceAccountKeyId && !propertyId) missingParams = `${serviceAccountMsg} and ${propertyMsg}`;
  if (!serviceAccountKeyId) missingParams = `${serviceAccountMsg};
  if (!propertyId) missingParams = `${propertyMsg};
```

Reply:
Thanks for this feedback! I made these adjustments.